### PR TITLE
🐛 Makefile release target should properly clean the git tree

### PR DESCRIPTION
**Pull request by vincepri**
Signed-off-by: Vince Prignano <vincepri@vmware.com>

**What this PR does / why we need it**:
Fixes an issue where the development manifests weren't being properly generated, because the tree would be dirty by the time we get to call `make release` under capd.

/milestone v0.3.10

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -450,12 +450,10 @@ release: clean-release ## Builds and push container images using the latest git
...
 	$(MAKE) set-manifest-pull-policy PULL_POLICY=IfNotPresent TARGET_RESOURCE="./controlplane/kubeadm/config/manager/manager_pull_policy.yaml"
-	$(MAKE) release-manifests
-	# Release CAPD components and add them to the release dir
-	$(MAKE) -C test/infrastructure/docker/ release
-	cp test/infrastructure/docker/out/infrastructure-components.yaml $(RELEASE_DIR)/infrastructure-components-development.yaml
-	# Adds CAPD templates
-	cp test/infrastructure/docker/templates/* $(RELEASE_DIR)/
+	## Build the manifests
+	$(MAKE) release-manifests clean-git-release
+	## Build the development manifests
```
[the remark this answers is not in the document]
Comment (vincepri) [reply Makefile:455]: They seem logically separate, so I split them

[the remark this answers is not in the document]
Comment (fabriziopandini) [reply Makefile:455]: Sorry I wast clear.
My concern is that we are generating the docker manifest only when we call `make release`, instead IMO this manifest should be generated also when we call `make release-manifests`

[the remark this answers is not in the document]
Comment (vincepri) [reply Makefile:455]: That seems ok to me, `release-manifests` should remain separate in terms of responsibilities. When we call `make release-manifests`, it'll behave the same way here and in the capd one. For `make release` it makes sense to call both

**Review by fabriziopandini — COMMENTED** [review]
/approve
/lgtm

**Pull request merged**